### PR TITLE
refactor(analyses-to-deliver)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Refactoring of the function `analyses_to_deliver`

### Added

- test for the function `get_analysis_to_deliver_for_pipeline`

### Changed

- renamend `analyses_to_deliver` to `get_analyses_to_deliver`

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do  `cg upload all --pipeline`
- [ ] Do  `cg upload clinical-delivery`
- [ ] Analyses are shown on trailblazer

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/api/base.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 """All models aggregated in a base class"""
-from typing import Type
+from typing import Type, List
 
 from alchy import Query, ModelBase
 from dataclasses import dataclass
-
+from cg.store.filters.status_analysis_filters import apply_analysis_filter, AnalysisFilter
```
Comment (diitaz93) [inline cg/store/api/base.py:6]: I could not find any usages of these imported objects in this script

File: cg/cli/upload/utils.py [hunk 2]
```diff
@@ -15,6 +15,6 @@
 def suggest_cases_to_upload(status_db: Store, pipeline: Optional[Pipeline] = None) -> None:
     """Print a list of suggested cases to upload."""
     LOG.warning("Provide a case, suggestions:")
-    records = status_db.analyses_to_upload(pipeline=pipeline)[:MAX_ITEMS_TO_RETRIEVE]
+    records = status_db.get_analyses_to_upload(pipeline=pipeline)[:MAX_ITEMS_TO_RETRIEVE]
```
Comment (diitaz93) [inline cg/cli/upload/utils.py:18]: add type hint

File: tests/store/api/find/test_find_business_data_analysis.py [hunk 3]
```diff
@@ -59,4 +61,35 @@ def test_get_latest_microsalt_analysis_to_upload(
...
+
+def test_get_analyses(store_with_analyses_for_cases: Store, analysis_id: int = 1):
+    """Test all analyses can be returned."""
+    # GIVEN a database with an analysis and case
+
+    # WHEN fetching the analysis by id
+    analysis: List[Analysis] = store_with_analyses_for_cases.get_analyses()
+
+    # THEN one analysis should be returned
+    assert len(analysis) == store_with_analyses_for_cases._get_query(table=Analysis).count()
```
Comment (diitaz93) [inline tests/store/api/find/test_find_business_data_analysis.py:95]: The comments don't seem to match with the code, and the parameter analysis_id is not used

**Review by diitaz93 — APPROVED** [review]
Looks good 👍 just some comments on the tests

**Comment by ChrOertlin** [comment]
image

**Pull request merged**